docs/references/plugin-compatibility.md

```diff
@@ -0,0 +1,134 @@
+# Plugin Compatibility
+
+DB-less mode is the preferred choice for controller-managed Kong and Kong
+Enterprise clusters. However, not all plugins are available in DB-less mode.
+Review the table below to check if a plugin you wish to use requires a
+database.
+
+Note that some DB-less compatible plugins [have some limitations or require
+non-default configuration for
+compatibility](https://docs.konghq.com/latest/db-less-and-declarative-config/#plugin-compatibility).
+
+## Kong
+
+|Plugin                          |  Kong                |  Kong (DB-less)    |
+|--------------------------------|----------------------|--------------------|
+|                                |  :white_check_mark:  |  :white_check_mark:|
+|acl                             |  :white_check_mark:  |  :white_check_mark:|
+|aws-lambda                      |  :white_check_mark:  |  :white_check_mark:|
+|azure-functions                 |  :white_check_mark:  |  :white_check_mark:|
+|basic-auth                      |  :white_check_mark:  |  :white_check_mark:|
+|bot-detection                   |  :white_check_mark:  |  :white_check_mark:|
+|correlation-id                  |  :white_check_mark:  |  :white_check_mark:|
+|cors                            |  :white_check_mark:  |  :white_check_mark:|
+|datadog                         |  :white_check_mark:  |  :white_check_mark:|
+|file-log                        |  :white_check_mark:  |  :white_check_mark:|
+|hmac-auth                       |  :white_check_mark:  |  :white_check_mark:|
+|http-log                        |  :white_check_mark:  |  :white_check_mark:|
+|ip-restriction                  |  :white_check_mark:  |  :white_check_mark:|
+|jwt                             |  :white_check_mark:  |  :white_check_mark:|
+|key-auth                        |  :white_check_mark:  |  :white_check_mark:|
+|kubernetes-sidecar-injector     |  :white_check_mark:  |  :white_check_mark:|
+|oauth2                          |  :white_check_mark:  |  :x:               |
+|prometheus                      |  :white_check_mark:  |  :white_check_mark:|
+|rate-limiting                   |  :white_check_mark:  |  :white_check_mark:|
+|request-termination             |  :white_check_mark:  |  :white_check_mark:|
+|request-transformer             |  :white_check_mark:  |  :white_check_mark:|
+|response-ratelimiting           |  :white_check_mark:  |  :white_check_mark:|
+|response-transformer            |  :white_check_mark:  |  :white_check_mark:|
+|syslog                          |  :white_check_mark:  |  :white_check_mark:|
+|tcp-log                         |  :white_check_mark:  |  :white_check_mark:|
+|udp-log                         |  :white_check_mark:  |  :white_check_mark:|
+|zipkin                          |  :white_check_mark:  |  :white_check_mark:|
+|application-registration        |  :x:                 |  :x:               |
```

Comment:
Let's remove all Enterprise-only plugins from this table since it compares OSS only options.